### PR TITLE
[DOC] README: Revise for new Consulting blog flow & general edits

### LINKED ISSUE #702: [DOC]: Update README with instructions for creating Consulting blogs
The [current Labs instructions] for blog post workflow are close in many details to the procedures/locations/etc. for the new GitHub-based Consulting blog workflow, but they are not identical.

This documentation in the README needs to be updated to accurately reflect the Consulting flow as well as the Labs flow.

[current labs instructions]: https://github.com/Quansight/Quansight-website#github-based-blog-workflow-labs-blog-

### PR DESCRIPTION
**Pull request by bskinn**
With the merge of #577, the Consulting blog is now operating on analogous GitHub-/Markdown-based machinery to the Labs blog.

The primary goal of this PR is to update the README to reflect this new Consulting blog workflow.

I also took the liberty of making some additional general edits to the README.

Closes #702.

### CODE REVIEW

**Comment by bskinn** [comment]
Would appreciate your review on these changes when you have a chance, @gabalafou -- thanks!

File: README.md [hunk 1]
```diff
@@ -504,30 +512,60 @@ You can fetch data from Storyblok directly using queries. To add the query:
...
+      post with one or more categories
+    - Converted to all caps for use in the category-selection dropdown on the
+      `/library` page
+  - `value`
+    - Should be written in slugified form and in all lowercase
+      - NOTE: Currently one category, `Infrastructure & HPC`, has a capitalized
+        `value` due to an oversight. We plan to fix this.
+    - Used as-is (**case sensitive**) in the `category` query parameter to
+      `/library`—URLs with mismatched case in the `category` query parameter
+      will return **no results**
```
Comment (gabalafou) [inline README.md:561]: Is this level of detail appropriate here. I know, I know, this whole README is overly detailed. But honestly, this explanation of the `categories.json` files makes my head spin. I hate the fact that JSON doesn't support comments.

I think this is valuable information. I'm just wondering if we could expose it somewhere more useful. Maybe not. Just wondering.

I think what this section is really screaming is that we need to simplify/fix the system. So maybe it would make sense to move all of this detail into an issue and link to the issue from the README?

File: README.md [hunk 2]
```diff
@@ -504,30 +512,60 @@ You can fetch data from Storyblok directly using queries. To add the query:
...
+
+- For the Labs blog, `categories.json` is an array of case-sensitive strings.
+  These strings are used:
+  - As-is in the rendered category list on the main `/blog` page
+  - In a custom slugified format (spaces replaced with `+`) when filtering
+    `/blog` by category
+    - For example, the category `"Beyond PyData"` is slugified as
+      `"Beyond+PyData`"
+    - This category filter is currently **case-sensitive**—URLs with mismatched
+      case in the `category` query parameter will return **no results**
```
Comment (gabalafou) [inline README.md:546]:
```suggestion
  - These strings appear as-is in the list of categories on the main `/blog` page.
  - When filtering `/blog` by category, they appear in a slugified format (spaces replaced with `+`), for example: `/blog?category=Beyond+PyData
  - Since these are case-sensitive, if a blog post includes a category with incorrect case in its metadata, then it will not show up under that category. And in the other direction, if the `category`  query parameter has incorrect case, it will return no results.
```

File: README.md [hunk 3]
```diff
@@ -582,28 +628,51 @@ hero:
...
-1. Create a new feature branch. Example: `feature/new-hello-world-post`.
-2. Add post feature and hero images to `apps/labs/public/posts/<post-name>`.
-3. Add a new `.md|.mdx` file inside the `app/labs/posts` directory. Make sure to read the
-   [Structure of the blog post section](#structure-of-the-blog-post) in this file to ensure
-   that the post is properly structured.
-4. Commit and push your changes to the repository. For commits please follow the format of the conventional commit.
-   [See](https://www.conventionalcommits.org/en/v1.0.0/)
-5. Create a pull request to the `develop` branch. Make sure to add the `type: content 📝` and `labs 🔭` labels to the PR.
-6. Wait for someone in the website team to review the new blog post. If everything is ok, the PR will be merged to the `develop` branch.
+> **Note** If there is a deadline or other timing consideration for the
```
Comment (gabalafou) [inline README.md:631]: Good note

File: README.md [hunk 4]
```diff
@@ -582,28 +628,51 @@ hero:
...
+> publication of your post, **please coordinate actively** with the Marketing
+> team (currently @Katebrack and @bskinn) and the website ops team
+> (#qwebsite-ops channel on Slack) from as early as possible to ensure we can
+> meet your requirements.
+
+1. Create a new feature branch off of the `develop` branch. Example:
+   `<site>-blog/new-hello-world-post`.
+2. Add post images (feature, hero, and any for the post body) to
+   `apps/<site>/public/posts/<post-name>`.
+3. Add a new `.md|.mdx` file inside the `app/<site>/posts` directory. Make sure
```
Comment (gabalafou) [inline README.md:641]:
```suggestion
3. Add a new `.mdx` file inside the `app/<site>/posts` directory. Make sure
```

As I mentioned in my other note, we should forget `.md`. It's especially misleading because in normal Markdown you can create code blocks but ["indented code does not work in MDX."](https://mdxjs.com/docs/what-is-mdx/#markdown)

File: README.md [hunk 5]
```diff
@@ -582,28 +628,51 @@ hero:
...
+   to read the
+   [Structure of the blog post section](#structure-of-the-blog-post) in this
+   file to ensure that the post is properly structured.
+4. Commit and push your changes to the repository. For commits please follow the
+   format of the conventional commit. See the
+   [Conventional Commits docs](https://www.conventionalcommits.org/en/v1.0.0/).
```
Comment (gabalafou) [inline README.md:647]:
```suggestion
   [Conventional Commit format](https://www.conventionalcommits.org/en/v1.0.0/).
```

I didn't actually know about this until today 😅

File: README.md [hunk 6]
```diff
@@ -582,28 +628,51 @@ hero:
...
+5. Create a pull request to the `develop` branch. Make sure to add the
+   `type: content 📝`, `owner: Quansight`, and either `labs 🔭` or
+   `consulting 🤝` labels to the PR.
+6. Wait for someone on the website team to review the new blog post. If
+   everything is ok, the PR will be merged to the `develop` branch. The blog
+   post will then go live with the next site build (merge to `main`).
```
Comment (gabalafou) [inline README.md:653]:
```suggestion
   post will then go live with the next production deployment (merge to `main`).
```

Since "production deployment" is the language that Vercel uses

File: README.md [hunk 7]
```diff
@@ -582,28 +628,51 @@ hero:
...
-4. Commit and push your changes to the repository. For commits please follow the format of the conventional commit.
-   [See](https://www.conventionalcommits.org/en/v1.0.0/)
-5. Wait for someone in the website team to review the new blog post. If everything is ok, the PR will be merged to the `develop` branch.
+2. Open the `apps/<site>/posts/categories.json` file.
+3. Add a new category/category object to the array. Make sure to follow the same
+   format as the other categories.
+4. Commit and push your changes to the repository. For commits please follow the
+   format of the conventional commit.
+   See the
+   [Conventional Commits docs](https://www.conventionalcommits.org/en/v1.0.0/).
```
Comment (gabalafou) [inline README.md:664]:
```suggestion
   [Conventional Commits format](https://www.conventionalcommits.org/en/v1.0.0/).
```

**Review by gabalafou — CHANGES_REQUESTED** [review]
My inner librarian heart is warmed by these README updates, thank you! 

I don't think any of my suggestions are blocking, but you might want to incorporate some of them, and we have to fix the merge conflict (banner vs overlay) before we can merge this.

File: README.md [hunk 8]
```diff
@@ -174,7 +176,7 @@ These are the concrete steps to follow to move your code from branch to branch:
    `develop` branch.
    - Consider doing a
      [squash-merge](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/incorporating-changes-from-a-pull-request/about-pull-request-merges#squash-and-merge-your-pull-request-commits),
-     especially if your pull request is relatively small, to keep a
+     except if your pull request is relatively small, to keep a
```
[the remark this answers is not in the document]
Comment (bskinn) [reply README.md:179]: Good point -- so, I'd think we'd want to squash all merges to `develop`, then.

My reasoning for squashing aggressively is because we're working on GitHub, where the original chain of commits is retained in the closed PR, even when what gets merged to the base branch is a single squash.

It has the benefit of clean base branch history, while still retaining the full arc of the work process in the PR.

So, I'm inclined to strengthen this to something like "Please use GitHub's squash-merge functionality for all PRs to `develop`."

---


For merges to `main`, since release/hotfix branches can have weird history topology (e.g., a release gets hotfixed, then merges to `main`, then merges to `develop`) that would be hard to follow if it got squash-merged to either `main` or `develop`, I'm inclined to do simple commit merges there.

File: README.md [hunk 9]
```diff
@@ -402,27 +409,28 @@ editors to previewing draft content against the `develop` branch to help catch
 any potential code/content conflicts before merging to `main`.
 
 When viewing the site in local development or at a Vercel preview URL, there
-should be a banner that explains that you are looking at a preview of the site.
-This helps reduce confusion when screenshots are shared. This banner should not
-be present on a production build of the site.
+should be a gray or yellow banner that indicates that you are looking at a
```
[the remark this answers is not in the document]
Comment (bskinn) [reply README.md:412]: This PR was opened well before that change in language. I'll need to rebase/merge from `develop` & deal with any conflicts before this is ready to merge.

File: README.md (hunk 4 above)
Comment (bskinn) [reply README.md:641]: I think all of the posts for both blogs are `.md` files. Should we rename them? Not great to have the churn in the file history.

If nothing else, any blogs not yet on `develop` should be renamed to `.mdx`?

File: README.md [hunk 10]
```diff
@@ -582,28 +628,51 @@ hero:
...
+5. Create a pull request to the `develop` branch. Make sure to add the
+   `type: content 📝`, `owner: Quansight`, and either `labs 🔭` or
+   `consulting 🤝` labels to the PR.
+6. Wait for someone on the website team to review the new blog post. If
+   everything is ok, the PR will be merged to the `develop` branch. The blog
+   post will then go live with the next site build (merge to `main`).
 
 ### Adding a new blog category
 
 1. Create a new feature branch.
```
[the remark this answers is not in the document]
Comment (bskinn) [reply README.md:657]: I'm solidly +1 to this, new categories only being created when posts are created that need them. I'll rewrite a draft for you to review.

Related, see #713.

File: README.md [hunk 11]
```diff
@@ -37,13 +37,21 @@ Here is some basic info to help orient you to this repo.
   - `./libs` holds code shared by both websites.
 - The websites' **content** lives in [Storyblok](https://app.storyblok.com)
   (requires login).
-  - But Labs and Consulting **blog posts** live under `./apps/labs/posts` and 
+  - But Labs and Consulting **blog posts** live under `./apps/labs/posts` and
     `./apps/consulting/posts`, respectively.
+- **Static content** for each site lives under `./apps/labs/public` and
+  `./apps/consulting/public`, respectively.
+  - The `public` path element is removed when the site is built: for example, a
```
[the remark this answers is not in the document]
Comment (bskinn) [reply README.md:44]: I've seen it ~50/50 with `/static` being the magic folder name for static content. It took me a decent while with the codebase to realize it was doing the same thing.

File: README.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (bskinn) [reply README.md:179]: I'll write out my case in a new commit here, see what you think on your next review pass.

File: README.md (hunk 9 above)
[the remark this answers is not in the document]
Comment (bskinn) [reply README.md]: `develop` merged in, conflicts fixed, and straggling instances of "banner" converted to "overlay" in this branch.

Should be taken care of on next commit.

File: README.md [hunk 12]
```diff
@@ -402,27 +409,28 @@ editors to previewing draft content against the `develop` branch to help catch
...
-be present on a production build of the site.
+should be a gray or yellow banner that indicates that you are looking at a
+preview of the site. This helps reduce confusion when screenshots are shared.
+This banner should not be present on a production build of the site.
 
 The preview/development environment banner should allow the user to toggle
 between seeing published versus draft/saved Storyblok content. The banner should
 change colors to indicate which of the two content preview modes the user is in
-(that is, whether they are seeing draft or published content from Storyblok).
+(that is, whether they are seeing draft (yellow banner) or published (gray
```
[the remark this answers is not in the document]
Comment (bskinn) [reply README.md:419]: Changed, will show in next commit.

File: README.md [hunk 13]
```diff
@@ -402,27 +409,28 @@ editors to previewing draft content against the `develop` branch to help catch
...
+banner) content from Storyblok).
```
[the remark this answers is not in the document]
Comment (bskinn) [reply README.md:420]: Same

File: README.md [hunk 14]
```diff
@@ -504,30 +512,60 @@ You can fetch data from Storyblok directly using queries. To add the query:
 3. Create the data retrieval function in the `/apps/.../api/utils` folder using the
    function, type, and hook created by the code gen script.
 
-## GitHub-based blog workflow (Labs blog) 💻
+## GitHub-based blog workflow (Labs and Consulting blogs) 💻
 
-All the **Quansight Labs** blog posts are located inside `apps/labs/post`,
-and therefore, any new posts _must_ be added to this same folder.
+All the blog posts for **both websites** are located inside `./apps/labs/posts`
```
[the remark this answers is not in the document]
Comment (bskinn) [reply README.md:517]: Will be de-emphasized in next commit.

File: README.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (gabalafou) [reply README.md:179]: +1 to all of that

File: README.md (hunk 4 above)
Comment (bskinn) [reply README.md:643]: If it would be good to rename the `.md` blog source files to `.mdx`, a good time to do that might be as part of #746, since we're touching all those files to add the categories `[]` anyways?

Comment (bskinn) [reply README.md:643]: I made this change locally, will reflect on next push

File: README.md (hunk 5 above)
Comment (bskinn) [reply README.md:649]: Made this change in both places it appears. Also capitalized 'Conventional Commit' in the flow of text, both places.

**Comment by bskinn** [comment]
@gabalafou Ok, I've made edits per your requests. Ready for your next pass.

Once you're satisfied with the content, I want to make a final text reflow pass over `README.md`, if that's all right with you.

**Comment by bskinn** [comment]
@gabalafou Annoying. The changes to the three blogs are due to automatic linting of staged files that husky does when I commit. I can't revert those files without figuring out how to temporarily disable that linting. 

&lt;googles&gt;

Nice, [`--no-verify` does work to skip Husky linting](https://github.com/typicode/husky/issues/124), good to know. :+1:

**Comment by bskinn** [comment]
Hm. With the migration to Wordpress complete, we should probably make note here of the `consulting` Jamstack site being idle, and that all Consulting site matters route through Marketing/Design...
